pc/protocols/usb_boot.c

```diff
@@ -324,14 +332,34 @@ usbBootError_t usb_find_device_with_bcd(unsigned idx, char *input_addr,
                         fprintf(stderr, "Found Address: %s - VID/PID %04x:%04x\n",
                                 input_addr, desc.idVendor, desc.idProduct);
                     }
-
                     libusb_ref_device(dev);
                     libusb_free_device_list(devs, 1);
                     if (bcdusb)
                         *bcdusb = desc.bcdUSB;
                     *device = dev;
                     devs = 0;
 
+                    int speed = libusb_get_device_speed(dev);
+                    char *speed_str[] = {"Unknown", "Low/1.5Mbps", "Full/12Mbps", "High/480Mbps", "Super/5000Mbps"};
+                    // if(speed != NULL)
+                    //     mv_strcpy(usb_speed, 15 ,speed_str[speed]);
+
+                    libusb_device_handle *dev_handle;
+                    if (libusb_open(dev, &dev_handle) == 0) {
+                        unsigned char sn[128];
+                        if (libusb_get_string_descriptor_ascii(dev_handle, desc.iSerialNumber, sn, sizeof sn) < 0)
+                            printf("Failed to get string descriptor\n");
+                        else
+                            printf("VID:%04x PID:%04x address:%s serial:%s Speed:%s\n",
+                                    desc.idVendor, desc.idProduct, input_addr, sn, speed_str[speed]);
+                        libusb_close(dev_handle);
+                        if(dev_conn_info != NULL){
+                            mv_strcpy(dev_conn_info, 15 ,speed_str[speed]);
+                            mv_strcpy(dev_conn_info + 15, 128 ,sn);
+                            printf("Speed is: %s \n", dev_conn_info );
```

Comment:
Will this always be printed? Use mvLog functionality rather, as otherwise the applications using this library cannot restrict this output

Reply:
it will be printed two times one on boot and another on connect. Will change it. Thanks